fv3core/stencils/xtp_u.py

```diff
@@ -123,37 +122,49 @@ def _compute_stencil(
         flux = _get_flux(u, courant, rdx, bl, br)
 
 
-def compute(c: FloatField, u: FloatField, flux: FloatField):
-    """
-    Compute flux of kinetic energy in x-dir.
-
-    Args:
-        c (in): Courant number in flux form
-        u (in): x-dir wind on D-grid
-        flux (out): Flux of kinetic energy
-    """
-    grid = spec.grid
-    iord = spec.namelist.hord_mt
-    if iord not in (5, 6, 7, 8):
-        raise NotImplementedError(
-            "Currently xtp_v is only supported for hord_mt == 5,6,7,8"
+class XTP_U:
+    def __init__(self, namelist):
+        iord = spec.namelist.hord_mt
+        if iord not in (5, 6, 7, 8):
+            raise NotImplementedError(
+                "Currently xtp_v is only supported for hord_mt == 5,6,7,8"
+            )
+        grid = spec.grid
+        self.origin = grid.compute_origin()
+        self.domain = grid.domain_shape_compute(add=(1, 1, 0))
+        ax_offsets = axis_offsets(spec.grid, self.origin, self.domain)
+        assert namelist.grid_type < 3
+        self.dxa = grid.dxa
```

Comment:
are we ever in a state where we have multiple of these objects with multiple namelists floating around? I am curious behind the reason to chose instance variables over class variables here?

Reply:
We currently don't have a use case where we need multiple namelists in the same thread, I think having global state for things that are essentially global for a run is intuitive and makes a lot of sense. But it does reduce the future potential flexibility to use the model and pieces of the model in isolated ways or for experiments where multiple namelists are used. I think when Jeremy presents the objects to us again we can definitely discuss more.